### PR TITLE
sui: contracts

### PR DESCRIPTION
**Pull request by jumpsiegel**

### CODE REVIEW

**Review by optke3 — APPROVED** [review]
lgtm!

**Comment by evan-gray** [comment]
@kcsongor can you help turn these into a graphite stack if you deem appropriate?

File: sui/token_bridge/sources/transfer_tokens.move [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        if (bridge_state::is_wrapped_asset<CoinType>(bridge_state)) {
+            // now we burn the wrapped coins to remove them from circulation
+            bridge_state::burn<CoinType>(bridge_state, coins);
+        } else {
+            // deposit native assets. this call to deposit requires the native
+            // asset to have been attested
+            bridge_state::deposit<CoinType>(bridge_state, coins);
+        };
+
+        // TODO - pending Mysten uniform token standard - figure out how to get normalization decimals for token and relayer fee amounts
```
Comment (kcsongor) [inline sui/token_bridge/sources/transfer_tokens.move:108]: this should be possible now

Comment (kcsongor) [reply sui/token_bridge/sources/transfer_tokens.move:108]: @optke3

File: sui/token_bridge/sources/bridge_state.move [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+    // Treasury caps, token stores, consumed VAAs, registered emitters, etc.
+    // are stored as dynamic fields of bridge state.
+    struct BridgeState has key, store {
+        id: UID,
+
+        /// Set of consumed VAA hashes
+        consumed_vaas: Set<vector<u8>>,
+
+        /// Token bridge owned emitter capability
+        emitter_cap: option::Option<EmitterCapability>,
```
Comment (kcsongor) [inline sui/token_bridge/sources/bridge_state.move:138]: we should make this a non-option

**Review by kcsongor — APPROVED** [review]
Approving with the understanding that we will still have to iterate on the contracts:
1. clear out TODOs
2. add documentation

but it's in a mergeable state now.

**Review by evan-gray — APPROVED** [review]
ack what @kcsongor said.

**Pull request merged**